internal/codespaces/ssh.go

```diff
@@ -95,9 +83,56 @@ func newSSHCommand(ctx context.Context, port int, dst string, cmdArgs []string)
 	return cmd, connArgs, nil
 }
 
-// parseSSHArgs parses SSH arguments into two distinct slices of flags and command.
-// It returns an error if a unary flag is provided without an argument.
 func parseSSHArgs(args []string) (cmdArgs, command []string, err error) {
+	return parseArgs(args, "bcDeFIiLlmOopRSWw")
+}
+
+// newSCPCommand populates an exec.Cmd to run an scp command for the files specified in cmdArgs.
+// cmdArgs is parsed such that scp flags and the files to copy are separated by a "--" in the command.
+// For example: scp -F ./config -- local/file remote:file
+func newSCPCommand(ctx context.Context, port int, dst string, cmdArgs []string) (*exec.Cmd, error) {
+	// Beware: invalid syntax causes scp to exit 1 with
+	// no error message, so don't let that happen.
+	connArgs := []string{
+		"-P", strconv.Itoa(port),
+		"-o", "NoHostAuthenticationForLocalhost=yes",
+		"-C", // compression
+	}
+
+	cmdArgs, command, err := parseSCPArgs(cmdArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	cmdArgs = append(cmdArgs, connArgs...)
+
+	cmdArgs = append(cmdArgs, "--")
```

Comment:
I wish there was a better place to put this, feels a bit out of nowhere. Perhaps a comment will help settle it in place?

Reply:
I realized it's not actually even required so I just removed it (updated comments accordingly)